server/Dockerfile

```diff
@@ -1,4 +1,18 @@
-FROM python:3.11.2
+FROM python:3.11.2-slim
+
+ARG TARGETPLATFORM
+
+# On platform arm/v7, `gcc`, `libc6-dev`, and `libffi-dev` must be installed to
+# be able to compile the `cffi` Python indirect dependency.
+RUN if [ "$TARGETPLATFORM" = "linux/arm/v7" ]; then \
+    apt-get update && \
+    apt-get install -y --no-install-recommends \
+        gcc \
+        libc6-dev \
+        libffi-dev \
```

Comment:
Nice work, however, a better approach would most likely be to include all of this in the RUN command on line 22, then you can also remove the build dependencies before exiting the method ([inspiration](https://github.com/JonasAlfredsson/docker-nginx-certbot/blob/master/src/Dockerfile#L45)).

Also, do you need to install and keep the [libffi7](https://github.com/JonasAlfredsson/docker-nginx-certbot/blob/master/src/Dockerfile#L29) package afterwards?

Reply:
I see you prioritize a reduced amount of layers and container size, over maintaining a better layer separation to get better Docker caching use, is that right? I would assume that all the heavy lifting for Docker build is always on GitHub actions, so needing to reinstall these apt-get dependencies every time `requirements.txt` gets updated is not as relevant.

Reply:
Regarding whether to keep libffi7 or not, I don't have an ARMv7 device to test if keeping it is needed. I'll follow your same pattern from `docker-nginx-certbot`.

Reply:
You are correct :)
Since this is such a simple and straight forward Dockerfile I believe we gain more by keeping the number of layers as few and small as possible. If we now are chasing the smallest image we will lose a bit by installing unnecessary (after the build is complete) dependencies. 
While you absolutely have a valid point about trying to reuse the cache, the github runners will start from a blank state and only have the base image as a common cache. 
If we would be keeping any of these dependencies after we are done with the pip install I would probably place them in a separate layer, but these transient ones are better to install and discard so our users won't even know about it :)

Yeah, same here... Let's include that one and see if someone screams :P

Reply:
I think the last iteration looks really good, will just make sure it builds then we merge :) 👍

Reply:
Great, thanks for the detailed answer! Makes complete sense :) 

I'm still getting a 196MB image for `devpi-server` on x86_64 after this dependency install/cleanup, and all platforms should build correctly now!